runtimes/java-18.0/src/main/java/io/openruntimes/java/Server.java

```diff
@@ -9,53 +9,158 @@
+
+        if(port != Integer.parseInt(defaultPort)) {
+            url += ":" + String.valueOf(port);
         }
 
-        Wrapper codeWrapper = new Wrapper();
-        RuntimeRequest request = new RuntimeRequest(req);
-        RuntimeResponse response = new RuntimeResponse();
+        url += path;
+
+        if(!queryString.isEmpty()) {
+            url += "?" + queryString;
+        }
+
+        RuntimeRequest runtimeRequest = new RuntimeRequest(bodyString, body, headers, method, url, scheme, host, path, port, queryString, query);
+        RuntimeResponse runtimeResponse = new RuntimeResponse();
+        RuntimeContext context = new RuntimeContext(runtimeRequest, runtimeResponse);
 
-        ByteArrayOutputStream outStream = new ByteArrayOutputStream();
-        ByteArrayOutputStream errStream = new ByteArrayOutputStream();
-        PrintStream userOut = new PrintStream(outStream);
-        PrintStream userErr = new PrintStream(errStream);
         PrintStream systemOut = System.out;
         PrintStream systemErr = System.err;
+        
+        ByteArrayOutputStream customstdStream = new ByteArrayOutputStream();
+        PrintStream customstd = new PrintStream(customstdStream);
+        System.setOut(customstd);
+        System.setErr(customstd);
 
-        System.setOut(userOut);
-        System.setErr(userErr);
+        RuntimeOutput output = null;
 
         try {
-            RuntimeResponse userResponse = codeWrapper.main(request, response);
-            Map<String, Object> output = new HashMap<String, Object>();
-            output.put("response", userResponse.data);
-            output.put("stdout", outStream.toString());
-            return resp.result(RuntimeResponse.gson.toJson(output));
+            Class classToLoad = Class.forName("io.openruntimes.java.Tests");
+            Method classMethod = classToLoad.getDeclaredMethod("main", RuntimeContext.class);
+            Object instance = classToLoad.newInstance();
+            output = (RuntimeOutput) classMethod.invoke(instance, context);
         } catch (Exception e) {
-            e.printStackTrace();
-            Map<String, Object> output = new HashMap<String, Object>();
-            output.put("stdout", outStream.toString());
-            output.put("stderr", errStream.toString());
-            return resp.code(500).result(RuntimeResponse.gson.toJson(output));
+            StringWriter sw = new StringWriter();
+            PrintWriter pw = new PrintWriter(sw);
+            e.printStackTrace(pw);
+
+            context.error(sw.toString());
+            output = context.res.send("", 500, new HashMap<String, String>());
         } finally {
             System.out.flush();
             System.err.flush();
             System.setOut(systemOut);
             System.setErr(systemErr);
         }
+
+        if(output == null) {
+            context.error("Return statement missing. return context.res.empty() if no response is expected.");
```

Comment:
If `output` is null here the function must have returned `null`; the message will be misleading

Reply:
I am making myself note to change the message in all runtimes. Let's ignore it as part of this PR